### PR TITLE
實作最後一人離開房間後則自動關閉房間

### LINKED ISSUE #139: 最後一人離開房間後則自動關閉房間
```
Given:
Only one Player "A" in Room "01"

When:
Player "A" left the Room "01"

Then:
Room "01" is closed automatically and removed from Room List
```

### PR DESCRIPTION
**Pull request by ted791029**
## Why need this change? / Root cause: 
- Readiness oriented room host
## Changes made:
- Room related
## Test Scope / Change impact:
- RoomController
## Issue
- resolve #139

### CODE REVIEW

File: application/src/main/kotlin/tw/waterballsa/gaas/application/usecases/LeaveRoomUsecase.kt [hunk 1]
```diff
@@ -17,6 +17,7 @@ class LeaveRoomUsecase(
             val player = findPlayerByIdentity(userIdentity)
             room.leaveRoom(player.id)
             roomRepository.leaveRoom(room)
+            if (room.isEmpty()) roomRepository.deleteById(room.roomId!!)
```
Comment (Wally5077) [inline application/src/main/kotlin/tw/waterballsa/gaas/application/usecases/LeaveRoomUsecase.kt:20]:
```kotlin
room.leaveRoom(player)

if (room.isEmpty()) {
    roomRepository.deleteById(room)
} else {
    roomRepository.leaveRoom(room)
}
```

**Review by Wally5077 — CHANGES_REQUESTED** [review]
**Review done.**

File: application/src/main/kotlin/tw/waterballsa/gaas/application/usecases/LeaveRoomUsecase.kt [hunk 2]
```diff
@@ -16,7 +16,12 @@ class LeaveRoomUsecase(
             val room = findRoomById(roomId)
             val player = findPlayerByIdentity(userIdentity)
             room.leaveRoom(player.id)
-            roomRepository.leaveRoom(room)
+
+            if (room.isEmpty()) {
+                roomRepository.deleteById(room.roomId!!)
```
Comment (Wally5077) [inline application/src/main/kotlin/tw/waterballsa/gaas/application/usecases/LeaveRoomUsecase.kt:21]: 昨天忘了補充，請幫我將 **deleteById** 改成下方這樣 🙏 
```kotlin
roomRepository.closeRoom(room)
```

**Review by Wally5077 — CHANGES_REQUESTED** [review]
**Review done.**

File: application/src/main/kotlin/tw/waterballsa/gaas/application/usecases/LeaveRoomUsecase.kt [hunk 3]
```diff
@@ -16,7 +16,12 @@ class LeaveRoomUsecase(
...
+            } else {
+                roomRepository.leaveRoom(room)
+            }
```
Comment (frankvicky) [inline application/src/main/kotlin/tw/waterballsa/gaas/application/usecases/LeaveRoomUsecase.kt:24]: NIT：可以考慮用 `when`

**Review by Wally5077 — APPROVED** [review]
**LGTM**

**Pull request merged**